app/assets/scripts/components/edit-map.js

```diff
@@ -27,10 +29,21 @@ const EditMap = React.createClass({
     this.addDraw();
 
     this.map.on('load', () => {
-      this.addEditLayer();
+      const prevAOI = this.props.geometry;
+      prevAOI.geometry.coordinates[0]
+        ? this.loadExistingSource(prevAOI)
+        : this.addNewSource();
     });
   },
 
+  componentWillReceiveProps: function (nextProps) {
+    const nextAOI = nextProps.geometry;
+    if (nextAOI.geometry.coordinates[0]) {
```

Comment:
Maybe it's a good idea to check if the next geometry is different than the previous. To avoid re-renders